querier.cc

```diff
@@ -391,6 +392,14 @@ std::optional<shard_mutation_querier> querier_cache::lookup_shard_mutation_queri
 }
 
 future<> querier_base::close() noexcept {
+    if (_qr_config) {
+        if ((*_qr_config).tombstone_failure_threshold > 0 && _qr_stats.dead >= (*_qr_config).tombstone_failure_threshold) {
+            qrlogger.error("Read {} live rows and {} tombstones for query {} {} (see tombstone_failure_threshold)", _qr_stats.live, _qr_stats.dead, (*_qr_config).query_raw_string, (*_range.get()));
+        } else if ((*_qr_config).tombstone_warn_threshold > 0 && _qr_stats.dead >= (*_qr_config).tombstone_warn_threshold) {
+            qrlogger.warn("Read {} live rows and {} tombstones for query {} {} (see tombstone_warn_threshold)", _qr_stats.live, _qr_stats.dead, (*_qr_config).query_raw_string, (*_range.get()));
+        }
```

Comment:
I don't think this distinction is valuable. A meaningful distinction would be logging vs. failing the query, but I'm not sure we want that, we already gracefully handle long strings of tombstones.

Reply:
I don't think this is the right place for this: this will be called only once when the querier is destroyed. This might be at the very end of the query (after all pages are read) or after N pages, when the querier is evicted.
I think we want to either check at the end of each page, or at the very end.

Reply:
I removed error message and only leave warn, and fix to show per each page or at the very end.